homeassistant/components/cover/mqtt.py

```diff
@@ -279,21 +279,26 @@ def position_message_received(topic, payload, qos):
             if self._template is not None:
                 payload = self._template.async_render_with_possible_json_value(
                     payload)
+
             if payload.isnumeric():
-                if 0 <= int(payload) <= 100:
-                    percentage_payload = int(payload)
+                int_payload = int(payload)
+                if (self._position_open == DEFAULT_POSITION_OPEN
+                        and self._position_closed == DEFAULT_POSITION_CLOSED
+                        and 0 <= int_payload <= 100):
+                    percentage_payload = int_payload
                 else:
                     percentage_payload = self.find_percentage_in_range(
                         float(payload), COVER_PAYLOAD)
                 if 0 <= percentage_payload <= 100:
```

Comment:
Why do we have this check here? Shouldn't we cap `percentage_payload` in `find_percentage_in_range`?

Reply:
I added this check because of users who have exactly opposite way of covers open=0,close=100, so slider would work always same way:
100% filled is full open, 0% filled is closed.
If we will be always calling `find_percentage_in_range`, then it shouldn't matter anymore.

Reply:
I think we also should cap the end points of the percentage range to 0 and 100 in `find_percentage_in_range`.

Reply:
Do you mean something like this:

```
def find_percentage_in_range(self, position, range_type=TILT_PAYLOAD):
        """Find the 0-100% value within the specified range."""
        # the range of motion as defined by the min max values
        if range_type == COVER_PAYLOAD:
            max_range = self._position_open
            min_range = self._position_closed
        else:
            max_range = self._tilt_max
            min_range = self._tilt_min
        current_range = max_range - min_range
        # offset to be zero based
        offset_position = position - min_range
        position_percentage = round(
            float(offset_position) / current_range * 100.0)
        if range_type == TILT_PAYLOAD and self._tilt_invert:
            return 100 - position_percentage
        if position_percentage > 100:
            position_percentage = 100
        elif position_percentage < 0:
            position_percentage = 0
        return position_percentage
```

Reply:
Use `min` and `max` and probably do it for tilt payload too.

Reply:
but max_range and min_range could be different values not percentage_position like 30 - 110.
I can add variables like this.
```
def find_percentage_in_range(self, position, range_type=TILT_PAYLOAD):
        """Find the 0-100% value within the specified range."""
        # the range of motion as defined by the min max values
        max = 100
        min = 0
        if range_type == COVER_PAYLOAD:
            max_range = self._position_open
            min_range = self._position_closed
        else:
            max_range = self._tilt_max
            min_range = self._tilt_min
        current_range = max_range - min_range
        # offset to be zero based
        offset_position = position - min_range
        position_percentage = round(
            float(offset_position) / current_range * 100.0)
        if position_percentage > max:
            position_percentage = max
        elif position_percentage < min:
            position_percentage = min
        if range_type == TILT_PAYLOAD and self._tilt_invert:
            return 100 - position_percentage
        return position_percentage
```

IMO I don't see any reason why tilts has this substraction 100 - x.
If tilts has slider than it is working differently for regular tilts and inverted tilts.

Reply:
I meant that we should use the builtin functions `max` and `min` instead of `if ... elif`.
```py
max_percent = 100
min_percent = 0
position_percentage = min(max(position_percentage, min_percent), max_percent)
```

Reply:
Cool, never used it before in such situation.
So I think I can easily delete this if as well, as min/max functions are taking care of it.
```
if 0 <= percentage_payload <= 100:
```